Sources/ClientRuntime/Networking/Streaming/DataStreamReader.swift

```diff
@@ -42,15 +42,21 @@ public class DataStreamReader: StreamReader {
     }
     
     public func read(maxBytes: UInt? = nil, rewind: Bool = false) -> ByteBuffer {
-        let buffer = ByteBuffer(size: Int(maxBytes ?? availableForRead))
+        var data = Data()
         withLockingClosure {
-            buffer.put(byteBuffer, offset: offset, maxBytes: maxBytes)
-            if !rewind {
-                _availableForRead -= UInt(buffer.length)
-                offset += UInt(buffer.length)
+            let count = Int(maxBytes ?? availableForRead)
+            let ptr = UnsafeMutablePointer<UInt8>.allocate(capacity: count)
+            defer { ptr.deallocate() }
+            let buffer = UnsafeMutableBufferPointer(start: ptr, count: count)
+            let bytesRead = byteBuffer.read(buffer: buffer)
+            data = Data(buffer)
```

Comment:
We can replace all of this with data and a safer version which doesn't involve allocation/deallocation. Something like the following
```suggestion
            let data = Data(count: count)
            bufferData.withUnsafeMutableBytes { buffer in 
                 byteBuffer.read(buffer: buffer)
            }
```

Reply:
In general, the allocation & deallocation should only happen if we want to manually control lifetime of something. In most cases, we can avoid that.

Reply:
🤯 
awesome! will update to this